### PR TITLE
set INSTALL_INTERFACE to a relative ./include path rather than a hard path of CMAKE_INSTALL_PREFIX

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -257,8 +258,8 @@ target_include_directories(${PDAL_BASE_LIB_NAME}
         ${ZSTD_INCLUDE_DIRS}
         ${NLOHMANN_INCLUDE_DIR}
     PUBLIC
-        ${GDAL_INCLUDE_DIR}
-        ${LASZIP_INCLUDE_DIR}
+        $<BUILD_INTERFACE:${GDAL_INCLUDE_DIR}>
```
Comment (abellgithub) [inline CMakeLists.txt]: Why?

Comment (hobu) [reply CMakeLists.txt]: https://stackoverflow.com/a/25681179/498396

Comment (abellgithub) [reply CMakeLists.txt]: But then it seems pointless without an INSTALL_INTERFACE entry.  Also, I don't think these should be public anymore, but I didn't want to change it right before release.

Comment (hobu) [reply CMakeLists.txt]: > But then it seems pointless without an INSTALL_INTERFACE entry.

Those libraries don't expose _INTERFACE at all, so we would have to create those for laszip and GDAL too.

**Pull request closed without merging**